### PR TITLE
Remove bunch of stuff from ensemble config

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
Please consider the commit messages for explanations of what's going on, and justifications for the removals.

I will / intend to fixup / squash many of the commits before merging, they are mostly left isolated to make the review process easier.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/clib/lib/enkf/enkf_config_node.cpp [hunk 1]
```diff
@@ -691,29 +666,29 @@ enkf_config_node_type *enkf_config_node_alloc_GEN_DATA_from_config(
...
+                                     gen_data_file_format_type input_format,
+                                     const int_vector_type *report_steps) {
     enkf_config_node_type *config_node = NULL;
 
-    if ((init_file_fmt == NULL) && (ecl_file == NULL) &&
-        (result_file != NULL)) {
+    printf("%s: node_key = `%s`\n", __func__, node_key);
+
+    printf("%s: result_file = `%s`\n", __func__, result_file);
+    printf("%s: input_format = `%d`\n", __func__, input_format);
```
Comment (valentin-krasontovitsch) [inline src/clib/lib/enkf/enkf_config_node.cpp:679]: remove forgotten print statements!

Comment (valentin-krasontovitsch) [reply src/clib/lib/enkf/enkf_config_node.cpp]: done

**Comment by valentin-krasontovitsch** [comment]
if you should happen to review and want to merge during my absence, which i sincerely hope, feel free to squash all the commits, I think this can just be bundled up as remove unused code - maybe keep the bit i wrote about the export function only being used (according to both code and comment) for the field type, if you feel like it.

File: src/clib/lib/enkf/enkf_config_node.cpp [hunk 2]
```diff
@@ -691,29 +666,24 @@ enkf_config_node_type *enkf_config_node_alloc_GEN_DATA_from_config(
...
     enkf_config_node_type *config_node = NULL;
 
-    if ((init_file_fmt == NULL) && (ecl_file == NULL) &&
-        (result_file != NULL)) {
+    if (result_file != NULL) {
         config_node = enkf_config_node_alloc_GEN_DATA_result(
             node_key, input_format, result_file);
-    } else if ((init_file_fmt != NULL) && (ecl_file != NULL) &&
-               (result_file != NULL)) {
+    } else if (result_file != NULL) {
```
Comment (DanSava) [inline src/clib/lib/enkf/enkf_config_node.cpp:679]: This does not make a lot of sense now that `init_file_fmt` and  `ecl_file` have been removed

File: src/clib/lib/enkf/enkf_config_node.cpp [hunk 3]
```diff
@@ -646,20 +635,10 @@ enkf_config_node_type *enkf_config_node_alloc_GEN_DATA_from_config(
...
-                }
-
-                if ((init_file_fmt == NULL) && (ecl_file == NULL) &&
-                    (result_file != NULL))
+                if (result_file != NULL)
                     config_node = enkf_config_node_alloc_GEN_DATA_result(
                         node_key, input_format, result_file);
-                else if ((init_file_fmt != NULL) && (ecl_file != NULL) &&
-                         (result_file != NULL))
+                else if (result_file != NULL)
```
Comment (DanSava) [inline src/clib/lib/enkf/enkf_config_node.cpp:641]: This `else if` can be removed as well

File: src/clib/lib/enkf/enkf_node.cpp [hunk 4]
```diff
@@ -30,30 +30,30 @@
...
+  |  ==========|=====           |                =====================
+  |  |        \|/   |   o--------                |                   |
+  |  |  ================        |------------->  |                   |
+  |  |  |              |        |                |  field_config     |
+  |  |  |              |        |                |                   |
+  ===|  |  field       |  o------                |                   |
+  |     |              |                         |                   |
+  ===   |              |                         =====================
+        |              |
+        ================
```
Comment (DanSava) [inline src/clib/lib/enkf/enkf_node.cpp:56]: :rofl:

File: src/ert/_c_wrappers/enkf/data/gen_data.py [hunk 5]
```diff
@@ -31,14 +28,6 @@ def free(self):
     def __repr__(self):
         return f"GenData(len = {len(self)}) {self._ad_str()}"
 
-    def export(self, file_name, file_format_type, fortio):
```
Comment (DanSava) [inline src/ert/_c_wrappers/enkf/data/gen_data.py:34]: My IDE is telling me this export function is used in `src/ert/gui/tools/export/export_tool.py` not sure if that is true or not but worth double checking.

**Review by DanSava — COMMENTED** [review]
Overall looks good just some minor comments

File: src/clib/lib/enkf/enkf_node.cpp (hunk 4 above)
Comment (valentin-krasontovitsch) [reply src/clib/lib/enkf/enkf_node.cpp:56]: i couldn't help myself : D

File: src/ert/_c_wrappers/enkf/data/gen_data.py (hunk 5 above)
Comment (valentin-krasontovitsch) [reply src/ert/_c_wrappers/enkf/data/gen_data.py:34]: thank you, gonna investigate!

ya it's a bit so-so, i didn't go that deep, figured its gui so it's fine, checked it out, it defines an export function, and that one calls another export funtion - on something called `__export_widget()` - i just assumed that this widget is something QT-y, and not related to a gen data thingy ^^

and indeed, now looking a bit deeper, the `__export_widget` gets assigned a `ref` of something that inherits from a QT widget, and has an `export` function defined, which does not do anything with gen data.

this is why i cannot trust IDEs anymore ; )

**Review by DanSava — APPROVED** [review]
Looks good to me  :+1:

**Comment by sondreso** [comment]
Could the naming of this PR be updated to be a bit more informative for the release notes?

**Comment by valentin-krasontovitsch** [comment]
it is done!

**Pull request merged**